### PR TITLE
Adds notes for blank user_id and username

### PR DESCRIPTION
**Pull request by lamagnifica**
Resolves DOC-2413 based on AN-6137. 

@shnayder @stroilova : the first commit addresses this issue.
@srpearce 

tests ran without warnings or errors

### CODE REVIEW

File: en_us/data/source/internal_data_formats/credentials.rst [hunk 1]
```diff
@@ -8,6 +8,14 @@ EdX transfers course data to the data czars at our partner institutions in
 regularly generated data packages. Data packages can only be accessed by a
 single contact at each university, referred to as the "data czar".
 
+.. contents::
+  :local:
+  :depth: 2
+
+*************************************
+Key Set Up and Exchange Overview
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/credentials.rst]: Set Up (verb phrase) --> Setup (noun)

File: en_us/data/source/internal_data_formats/discussion_data.rst [hunk 2]
```diff
@@ -288,13 +300,16 @@ votes
   * count = total votes cast.
   * point = net vote, now always equal to up_count.
 
-A user only has one vote per ``Comment`` or ``CommentThread``. Though it's still written to the database, the UI no longer displays an option to downvote anything.
+A user only has one vote per ``Comment`` or ``CommentThread``. Though it's
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/discussion_data.rst]: "it's" --> "the vote" (grammatically, this sentence says the _UI_ "is still written to the database").

File: en_us/data/source/internal_data_formats/discussion_data.rst [hunk 3]
```diff
@@ -312,17 +330,24 @@ comment_count
...
 --------------------
-  A course team can attach a discussion to any piece of content in the course, or to top level categories like "General" and "Troubleshooting". When the discussion is a top level category it is specified in the course's policy file, and the ``commentable_id`` is formatted like this: "i4x-edX-edX101-course-How_to_Create_an_edX_Course". When the discussion is a specific component in the course, the ``commentable_id`` identifies that component: "d9f970a42067413cbb633f81cfb12604".
+  A course team can attach a discussion to any piece of content in the course,
+  or to top level categories like "General" and "Troubleshooting". When the
+  discussion is a top level category it is specified in the course's policy
+  file, and the ``commentable_id`` is formatted like this:
+  "i4x-edX-edX101-course-How_to_Create_an_edX_Course".
+  When the discussion is a specific component in the course, the
+  ``commentable_id`` identifies that component:
+  "d9f970a42067413cbb633f81cfb12604".
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/discussion_data.rst]: Suggested revision:

A course team can attach a discussion to any piece of content in the course, or to top level categories like "General" and "Troubleshooting". When the discussion is a top level category, it is specified in the course's policy file, and the `commentable_id` uses the format `i4x-institution-course_number-course-course_name` (for example, `i4x-edX-edX101-course-How_to_Create_an_edX_Course`). When the discussion is a specific component in the course, the `commentable_id` identifies that component (for example, `d9f970a42067413cbb633f81cfb12604`).

File: en_us/data/source/internal_data_formats/discussion_data.rst [hunk 4]
```diff
@@ -349,9 +374,13 @@ thread_type
...
 
-The following fields are specific to ``Comment`` objects. A ``Comment`` is either a response to a ``CommentThread`` (such as an answer to the question), or a reply to another ``Comment`` (a comment about somebody's answer).
+The following fields are specific to ``Comment`` objects. A ``Comment`` is
+either a response to a ``CommentThread`` (such as an answer to the question),
+or a reply to another ``Comment`` (a comment about somebody's answer).
 
-**History**: It used to be the case that ``Comment`` replies could nest much more deeply, but this was later capped at just these three levels (post, response, comment) much in the way that StackOverflow does.
+**History**: It used to be the case that ``Comment`` replies could nest much
+more deeply, but this was later capped at just these three levels (post,
+response, comment) much in the way that StackOverflow does.
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/discussion_data.rst]: In earlier versions of the edX platform, `Comment` replies could nest much more deeply. However, edX later restricted replies to three levels (post, response, and comment), similar to the practice on StackOverflow.

File: en_us/data/source/internal_data_formats/discussion_data.rst [hunk 5]
```diff
@@ -361,12 +390,17 @@ visible
 --------------------
 abuse_flaggers
 --------------------
-  Records the user id of each user who selects the **Report Misuse** flag for a ``Comment`` in the user interface. Stores an array of user ids if more than one user flags the ``Comment``. This is empty if no users flag the ``Comment``.
+  Records the user id of each user who selects the **Report Misuse** flag for a
```
Comment (srpearce) [inline en_us/data/source/internal_data_formats/discussion_data.rst]: Cap ID here and in the next line

**Comment by srpearce** [comment]
Just a few suggestions (I realize most of the changes here were spacing/line wrap issues, but found some text that could be more clear). Otherwise, :+1:

File: en_us/data/source/internal_data_formats/credentials.rst (hunk 1 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/credentials.rst]: Yes, this is an overview of the process of setting up and exchanging one's keys. I know it's awkward, but I do want to use the verbs here, not nouns.

File: en_us/data/source/internal_data_formats/discussion_data.rst (hunk 2 above)
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/discussion_data.rst]: it = the vote
I will rewrite (this is original text from >2 years ago. I think that rewriting/copy editing the entire file is outside the scope of this particular story.)

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 6]
```diff
@@ -226,6 +227,11 @@ The following member fields are present in the ``context`` field for all events.
      - number
      - Identifies the individual who is performing the action.
 
+.. note:: Occasionally, an event is recorded with a missing or blank
+ ``context.user_id`` value. This can occur when a user logs out, or the login
+ session times out, while a browser window remains open. Subsequent actions are
+ logged, but the system cannot supply the user identifier. EdX recommends that
+ you ignore these events during analysis.
```
Comment (lamagnifica) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:235]: @shnayder @stroilova this is the only part changed for the referenced DOC story. ^^

Comment (stroilova) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:235]: LGTM thanks @lamagnifica

**Pull request merged**